Controls/ViewSettings.ascx.vb

```diff
@@ -107,7 +108,21 @@ Namespace Controls
     End If
 
     ddBlogModuleId.Items.Clear()
-    ddBlogModuleId.DataSource = (New DotNetNuke.Entities.Modules.ModuleController).GetModulesByDefinition(PortalId, "DNNBlog.Blog")
+
+    Dim listOfValidBlogModules As List(Of ModuleInfo) = New List(Of ModuleInfo)
+    Dim tabController As DotNetNuke.Entities.Tabs.TabController = New DotNetNuke.Entities.Tabs.TabController()
+
+    For Each blogModule As ModuleInfo In (New DotNetNuke.Entities.Modules.ModuleController).GetModulesByDefinition(PortalId, "DNNBlog.Blog")
+     Dim blogPage As DotNetNuke.Entities.Tabs.TabInfo = tabController.GetTab(blogModule.TabID, PortalId, False)
+     blogModule.ModuleTitle = String.Concat(blogPage.TabName, " - ", blogModule.ModuleTitle)
+
+     Dim targetViewSetting As Common.ViewSettings = Common.ViewSettings.GetViewSettings(blogModule.TabModuleID)
+     If targetViewSetting IsNot Nothing AndAlso CBool(targetViewSetting.BlogModuleId = -1) Then
```

Comment:
Correct, I'm not filtering out non-configured modules, but that's a great way to do that. Adjusting now.